File: src/frontend/rate-my-fit/src/types/fit.ts
import { UserPreview } from "./user";
import { Item } from "./item";

export type Fit = {
  fitID: string,
  title: string,
  description: string | null,
  date: string,
  itemsID: string[],
  authorToken: string,
  picnames: string[],
  totalPrice: number,
  totalReviews: number,
  avgGrade: number,
  author: UserPreview,
  items: Item[]
};

export type Form = {
  title: string,
  description: string | null,
  pics: File[],
  itemsID: string[]
};

export type FormType = 'add' | 'update';

File: src/backend/models/fit.py
from typing import Optional

from models.abstract import AbstractModel
from models.user import UserCredentialsModel


class FitModel(AbstractModel):
    userCredentials: UserCredentialsModel
    fitID: Optional[str] = None
    title: str
    date: str
    description: Optional[str] = None
    itemsID: Optional[list[str]] = None
    totalPrice: int
    totalReviews: int
    avgGrade: float
